### ISSUE TITLE
Issue: Langsmith KeyError when tracing LCEL chain

### ISSUE THREAD

**Issue by quadcube** [langchain-ai/langsmith-sdk#752]
### Issue
The code that was run is as below. When Langsmith is disabled, the chain runs without a problem. 
For the agent routes, Langsmith works just fine.

```
class InputPlaygroundChat(BaseModel):
    """Input for the playground chat endpoint."""
    input: str
    messages: List[Union[HumanMessage, AIMessage, SystemMessage]] = Field(
        ...,
        description="The chat messages representing the current conversation.",
    )


"""
FastAPI app
"""
app = FastAPI(
    title="Simple API server",
    version="1.0",
    description="Simple API server",
)


prompt = ChatPromptTemplate.from_messages(
    [
        ("system", "You are an AI assistant that helps people to find information."),
        MessagesPlaceholder(variable_name="messages"),
    ]
)
chat_model = ChatOpenAI(model="gpt-3.5-turbo-0125", api_key="xxx")
chain = prompt | chat_model
add_routes(
    app,
    chain.with_types(input_type=InputPlaygroundChat),
    path="/openai",
    enable_feedback_endpoint=True,
    enable_public_trace_link_endpoint=True,
    playground_type="chat",
)
```

### Installation
Langchain 0.2.1
Langchain-core 0.2.3
Langchain-openai 0.1.6
Langserve 0.2.1
Langsmith 0.1.65

### Error trace
```
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/sse_starlette/sse.py", line 282, in __call__
    await wrap(partial(self.listen_for_disconnect, receive))
  File "/usr/local/lib/python3.11/site-packages/sse_starlette/sse.py", line 271, in wrap
    await func()
  File "/usr/local/lib/python3.11/site-packages/sse_starlette/sse.py", line 221, in listen_for_disconnect
    message = await receive()
              ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 553, in receive
    await self.message_event.wait()
  File "/usr/local/Cellar/python@3.11/3.11.7_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/locks.py", line 213, in wait
    await fut
asyncio.exceptions.CancelledError: Cancelled by cancel scope 10c9ff090

During handling of the above exception, another exception occurred:

  + Exception Group Traceback (most recent call last):
  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 399, in run_asgi
  |     result = await app(  # type: ignore[func-returns-value]
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
  |     return await self.app(scope, receive, send)
  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
  |     await super().__call__(scope, receive, send)
  |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__
  |     await self.middleware_stack(scope, receive, send)
  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 186, in __call__
  |     raise exc
  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__
  |     await self.app(scope, receive, _send)
  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 65, in __call__
  |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
  |     raise exc
  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
  |     await app(scope, receive, sender)
  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 756, in __call__
  |     await self.middleware_stack(scope, receive, send)
  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 776, in app
  |     await route.handle(scope, receive, send)
  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 297, in handle
  |     await self.app(scope, receive, send)
  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 77, in app
  |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
  |     raise exc
  |   File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
  |     await app(scope, receive, sender)
  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 75, in app
  |     await response(scope, receive, send)
  |   File "/usr/local/lib/python3.11/site-packages/sse_starlette/sse.py", line 268, in __call__
  |     async with anyio.create_task_group() as task_group:
  |   File "/usr/local/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 664, in __aexit__
  |     raise BaseExceptionGroup(
  | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/usr/local/lib/python3.11/site-packages/sse_starlette/sse.py", line 271, in wrap
    |     await func()
    |   File "/usr/local/lib/python3.11/site-packages/sse_starlette/sse.py", line 251, in stream_response
    |     async for data in self.body_iterator:
    |   File "/usr/local/lib/python3.11/site-packages/langserve/api_handler.py", line 1214, in _stream_log
    |     async for chunk in self._runnable.astream_log(
    |   File "/usr/local/lib/python3.11/site-packages/langchain_core/runnables/base.py", line 920, in astream_log
    |     async for item in _astream_log_implementation(  # type: ignore
    |   File "/usr/local/lib/python3.11/site-packages/langchain_core/tracers/log_stream.py", line 637, in _astream_log_implementation
    |     await task
    |   File "/usr/local/lib/python3.11/site-packages/langchain_core/tracers/log_stream.py", line 591, in consume_astream
    |     async for chunk in runnable.astream(input, config, **kwargs):
    |   File "/usr/local/lib/python3.11/site-packages/langchain_core/runnables/base.py", line 4620, in astream
    |     async for item in self.bound.astream(
    |   File "/usr/local/lib/python3.11/site-packages/langchain_core/runnables/base.py", line 2800, in astream
    |     async for chunk in self.atransform(input_aiter(), config, **kwargs):
    |   File "/usr/local/lib/python3.11/site-packages/langchain_core/runnables/base.py", line 2783, in atransform
    |     async for chunk in self._atransform_stream_with_config(
    |   File "/usr/local/lib/python3.11/site-packages/langchain_core/runnables/base.py", line 1863, in _atransform_stream_with_config
    |     context.run(_set_config_context, child_config)
    |   File "/usr/local/lib/python3.11/site-packages/langchain_core/runnables/config.py", line 126, in _set_config_context
    |     rt = RunTree.from_runnable_config(dict(config))
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/usr/local/lib/python3.11/site-packages/langsmith/run_trees.py", line 339, in from_runnable_config
    |     dotted_order = tracer.order_map[cb.parent_run_id][1]
    |                    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
    | KeyError: 'a3bc55e3-a991-49ec-b5a1-e4eb4626c819'
    +------------------------------------

```

**Comment by hinthornw** [comment]
Thanks for raising - seems related to some of the changes we just made to permit handoffs

**Comment by hinthornw** [comment]
Is this still ocurring in langsmith 0.1.66?

**Comment by hinthornw** [comment]
Will re-open until it is confirmed as fixed

**Comment by quadcube** [comment]
> Is this still ocurring in langsmith 0.1.67?

Thank you for the quick fix, I can confirm it work on langsmith 0.1.67! Cheers!
